app/_components/PoiCard.tsx

```diff
@@ -3,28 +3,43 @@ import { Button } from "./ui/button";
 import { useContext, useState } from "react";
 import { Pin } from "../_utils/global";
 import { AuthContext } from "./useContext/AuthContext";
-import { Coordinates } from "../_utils/coordinateMath";
+import {
+  Coordinates,
+  GetDistanceFromCoordinatesToMeters,
+} from "../_utils/coordinateMath";
 
 export function PoiCard({
   id,
   payload,
```

Comment:
Could `payload` have a more consistant name since the purpose of this component is `pin`.